bot/cogs/manage_classes_cog.py

```diff
@@ -1,365 +1,181 @@
-        role = await self.create_role(ctx, class_repr)
-
-        # Sleep here to make sure the role has been sent to the database and added
-        await asyncio.sleep(0.5)
-        msg = await ctx.send(f'!role add {role.id}')
-        await msg.delete(delay=1)
-
-        # sync perms with cleanup role
-        await self.sync_perms(ctx, channel, role)
-
-    async def input_class(self, ctx, class_repr: ClassType) -> ClassType | None:
-        def input_check(msg: discord.Message) -> bool:
-            return msg.author == ctx.author and ctx.channel == msg.channel
-
-        # check if the initial command contained a class abbv and number
-        if not class_repr.abbv:
-            embed = discord.Embed(
-                title="**New class setup wizard started :white_check_mark:**",
-                color=Colors.Purple,
-            )
-            avi = ctx.author.display_avatar.url
-            embed.set_footer(text=f"{ctx.author} - Time Limit: {TIMEOUT} Seconds", icon_url=avi)
-            embed.add_field(name="**Current values**", value=class_repr)
-            embed.add_field(
-                name="Please enter the class abbreviation and name E.G.",
-                value="cpsc-1010",
-                inline=False,
-            )
+    @app_commands.command(name='insert', description='Insert a class channel.')
+    async def insert(self, inter: discord.Interaction, channel: discord.TextChannel, role: discord.Role | None = None):
+        # check perms
+        if not await self._perms_check(inter):
+            return
+        # check if a current semester exists
+        if not await self.repo.get_current_semester():
+            embed = discord.Embed(title='📔 No Current Semester', color=Colors.Error)
+            embed.description = 'Could not insert class: there is no current semester in session.\n' \
+                                'Here is the information for the next upcoming semester.'
+            next_semester = await self.repo.get_next_semester()
+            embed.add_field(name='Next Semester', value=next_semester.semester_name)
+            embed.add_field(name='Start Date', value=as_timestamp(next_semester.start_date()))
+            await inter.response.send_message(embed=embed)
+            return
+        # check to see if the given channel is already in the repo
+        if await self.repo.search_class_by_channel(channel):
+            embed = error_embed(inter.user, f'The given channel {channel.mention} is already registered.')
+            await inter.response.send_message(embed=embed, ephemeral=True)
+            return
+        await inter.response.send_modal(AddClassModal(self.bot, channel=channel, role=role))
 
-            await ctx.send(embed=embed)
+    @app_commands.command(name='archive', description='Manually archive a class channel.')
+    async def class_archive(self, inter: discord.Interaction, channel: discord.TextChannel):
+        # check perms
+        if not await self._perms_check(inter):
```

Comment:
Can't we use d.py checks https://discordpy.readthedocs.io/en/stable/ext/commands/commands.html#checks I think they work with slash commands too

Reply:
Yes, I tested this out, but instead of something simple, like the bot telling the user they don't have permission, it will just raise an uncaught exception.

Reply:
Yes which is why you catch it :suspect: